arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -1325,6 +1328,160 @@
+        "connectionString"
+      ]      
+    },
+    "RoutingEventHubProperties": {
+      "description": "The properties related to an event hub endpoint.",
+      "type": "object",
+      "properties": {
+        "connectionString": {
+          "description": "The connectionstring of the event hub endpoint. ",
+          "type": "string"
+        },
+        "name": {
+          "description": "The name of the event hub endpoint. The endpoint name cannot be null or whitespace and should not start with a '$'. The endpoint name should not exceed 64 characters. The following names are reserved;  events, operationsMonitoringEvents, fileNotifications, $default. Endpoint names have to be unique across endpoint types.",
+          "type": "string",
+          "pattern": "^[A-Za-z0-9-._]{1,64}$"
+        },
+        "id": {
+          "description": "The identifier of the endpoint. This will be set by the server. Modifying this value causes create or update IoT hub operations to fail. Pass id as null when specifying a new endpoint. Pass the same id that you get for an endpoint from the service when specifying an existing endpoint during update operations.",
+          "type": "string",
+          "readOnly": true
+        },
+        "subscriptionId": {
+          "description": "The subscription identifier of the event hub endpoint.",
+          "type": "string"
+        },
+        "resourceGroup": {
+          "description": "The name of the resource group of the event hub endpoint.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "name",
+        "connectionString"
+      ]      
+    },    
+    "RouteProperties": {
+      "description": "The properties related to a routing rule based on which the IoT hub routes messages to endpoints.",      
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the route. The name has to be unique and cannot start with a $, also the route name should not exceed 64 characters.",      
+          "type": "string",
+          "pattern": "^[A-Za-z0-9-._]{1,64}$"
+        },
+        "source": {
+          "description": "The source to which the routing rule is to be applied to. e.g. DeviceMessages",      
+          "enum": ["DeviceMessages"],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "RoutingSource",
+            "modelAsString": true
+          }
+        },
+        "condition": {
+          "description": "The condition which is evaluated in order to apply the routing rule. If the condition is not provided it will evaluate to true by default. For grammar, See: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-query-language",      
+          "type": "string"
+        },
+        "endpointNames": {
+          "description": "The list of endpoints to which the messages that satisfy the condition are routed to. Currently only 1 endpoint is allowed.",
+          "minItems": 1,      
```

Comment:
Just curious: What are the reasons to model `endpointNames` as array of string that can only have 1 item?